src/context/config.tsx

```diff
@@ -34,13 +34,11 @@ export const useConfigContext = (): ConfigContext =>
   useContext<ConfigContext>(ConfigContext);
 
 export const ConfigContextProvider: FunctionComponent = ({ children }) => {
-  const [config, setConfig] = useState();
+  const [config, setConfig] = useState<Config>(DEFAULT_CONFIG);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
 
   const setConfigValue: ConfigContext["setConfigValue"] = (key, value) => {
-    const nextConfig = {
-      ...config,
-      [key]: value
-    };
+    const nextConfig = Object.assign({}, config, { [key]: value });
```

Comment:
Just noticed this, what's the reason for the change to `.assign`?

Reply:
`.assign` helps nextConfig inherit type Config, `...` spreader operation changes type to `object` which is not compatible with `setState<Config>`

Reply:
By using `.assign`, the `nextConfig` type becomes incorrect:
```
const nextConfig: Config & {
    [x: string]: string;
}
```
This accepts any key:value pairs.

Either way, we define `setConfigValue` such that the keys and values are restricted, so we are assured that `nextConfig` is of the correct type. We can also type it by doing the following, but there's no need as it's inferred from the param type of `setConfig`
```
const nextConfig: Config = {
  ...config,
  [key]: value
};
```

Reply:
Ahh okay got it 👍